### PR TITLE
update ansi_up to 3.0

### PR DESCRIPTION
**Pull request by minrk**
Fixes some ANSI processing issues (related to #3773)

- AnsiUp is a class now

<del>
- had to do some weird shenanigans to get the module to import (ansi_up has `exports = {default: AnsiUp}`)
  and I couldn't get it to import with anything but a require. Maybe someone with more typescript-foo than me can get it to work.
- ansi_up 2.0 *has* a typings file, but it isn't found. I copied the contents into ansi_up.d.ts and modified it to fit the 'declare module' structure.
- Fix missing `font-weight: bold` on bright ANSI foreground classes
</del>

Shenanigans fixed by AnsiUp 3.0

For an example, try:

!echo "foobar" | grep --color=always bar

which should color 'bar' as bold red. In master, it's plain black. It's correct with the CSS fixes and ansi_up upgrade here.

### CODE REVIEW

**Comment by jasongrout** [comment]
adding to beta 2, since it seems like this might be ready to go before we release beta 2 soon?

**Comment by minrk** [comment]
Not sure what the 'no name require' error is, since I've verified that it builds and works. I'll let someone else try to figure out how to get the module to import properly. It's a bit funky, but the module has:

```javascript
exports = { default: AnsiUp }
```

so the module itself has one member, called 'default', which is the AnsiUp class. If anyone knows how to import that, I couldn't figure it out without using require to circumvent spurious typing errors.

**Comment by ellisonbg** [comment]
I think I have wrestled with this type of things before (pun intended). This page was helpful:

https://www.typescriptlang.org/docs/handbook/modules.html

**Comment by ellisonbg** [comment]
It may be related to the `import module = require("module")` style syntax (sorry that page doesn't have links to sections)

File: packages/rendermime/style/index.css [hunk 1]
```diff
@@ -47,14 +47,14 @@
...
 
-.jp-RenderedText pre .ansi-bright-black-fg { color: #282C36; }
-.jp-RenderedText pre .ansi-bright-red-fg { color: #B22B31; }
-.jp-RenderedText pre .ansi-bright-green-fg { color: #007427; }
-.jp-RenderedText pre .ansi-bright-yellow-fg { color: #B27D12; }
-.jp-RenderedText pre .ansi-bright-blue-fg { color: #0065CA; }
-.jp-RenderedText pre .ansi-bright-magenta-fg { color: #A03196; }
-.jp-RenderedText pre .ansi-bright-cyan-fg { color: #258F8F; }
-.jp-RenderedText pre .ansi-bright-white-fg { color: #A1A6B2; }
+.jp-RenderedText pre .ansi-bright-black-fg { font-weight: bold; color: #282C36; }
```
Comment (mgeier) [inline packages/rendermime/style/index.css]: This is not an actual fix.
"Bright" colors can be used independently of "bold".
There should be a separate CSS class for "bold", but I guess this should be fixed in `ansi_up` directly?

**Comment by vidartf** [comment]
Version 3.0 of ansi_up was just released, including a fix for the typings, and changing how it treats bold (not sure if this is relevant to other discussions here).

File: packages/rendermime/style/index.css (hunk 1 above)
Comment (minrk) [reply packages/rendermime/style/index.css]: This has been fixed in ansi_up 3.0. Reverted this change here.

**Comment by minrk** [comment]
Updated PR to 3.0. Custom typings are now removed.

**Comment by vidartf** [comment]
The failing tests needs addressing:
```html
<span class="ansi-bright-green-fg ansi-red-bg">
```
changed to 
```html
<span style="font-weight:bold" class="ansi-green-fg ansi-red-bg">
```
with the new way of dealing with bold. The expected value of the tests should be updated (as long as the style attribute is acceptable).

**Comment by mgeier** [comment]
Now a bold font is used correctly, but using "bold" doesn't switch to the "bright" color anymore.
I think this PR as it is now is already an improvement, but there have to be several more fixes in ansi_up to get feature parity with the old Notebook.

**Pull request merged**